core/listener.go

```diff
@@ -96,11 +97,26 @@ func (cl *Listener) listen(ctx context.Context, sub <-chan *types.Block) {
 				return
 			}
 
-			eh, err := cl.construct(ctx, b, comm, vals, cl.bServ)
+			// extend block data
+			eds, err := extendBlock(b.Data)
+			if err != nil {
+				log.Errorw("listener: extending block data", "err", err)
+				return
+			}
+			// generate extended header
+			eh, err := cl.construct(ctx, b, comm, vals, eds)
 			if err != nil {
 				log.Errorw("listener: making extended header", "err", err)
 				return
 			}
+			// store block data if not empty
+			if eds != nil {
+				err = cl.store.Put(ctx, eh.DAH.Hash(), eds)
```

Comment:
do the edses of empty blocks end up here? I'm asking because Put will throw a `dagstore.ShardAlreadyExists` when the same DAH is put into the store twice

Reply:
good q